main.go

```diff
@@ -196,20 +244,20 @@ func main() {
 	println("")
 
 	var dbFilePath string
+	flag.UintVar(&idleSec, "idle", 60, "password timeout, in seconds (use 0 for no timeout)")
+	flag.StringVar(&dbFilePath, "db", getGoHashFilePath(), "file where password data is stored")
+	flag.Parse()
 
-	switch len(os.Args) {
-	case 1:
-		dbFilePath = getGoHashFilePath()
-	case 2:
-		dbFilePath = os.Args[1]
-		if !parentDirExists(dbFilePath) {
-			panic("The provided file is under a non-existing directory. Please create the directory manually first.")
-		}
-		if isDir(dbFilePath) {
-			panic("The path you provided is a directory. Please provide a file.")
-		}
-	default:
-		panic("Too many arguments provided. go-hash only accepts none or one argument: the passwords file.")
+	if !parentDirExists(dbFilePath) {
+		panic("The provided file is under a non-existing directory. Please create the directory manually first.")
+	}
+	if isDir(dbFilePath) {
+		panic("The path you provided is a directory. Please provide a file.")
+	}
+
+	if len(flag.Args()) > 0 {
+		fmt.Printf("usage: %s [-db <database filename>] [-idle <password timeout>]", os.Args[0])
```

Comment:
Hm... even though I like this change, I see no reason why the previous usage shouldn't be supported.
This is a kind of "breaking change" :) and I do not like to make breaking changes unless they have strong reasons.

Reply:
I added `-idle` flag first. Then, use of os.Args became problematic.  Could have used flag.Args() instead, I suppose.  But the direction I went with was to make the code treat both as a flag.  I think more idiomatic.  True, a breaking change, but it breaks rather nicely in that users will get a clear error message.

Reply:
Here's a suggestion you can of course take or leave...

Use command args to run go-hash in "one-shot" mode.  What I mean is, for example:

```
go-hash -db /some/file goto somegroup:someentry
```

go-hash would then prompt for master pass.  If correct, it would run command "goto somegroup:someentry", then exit immediately.  I personally would prefer the interactive mode to add and edit entries, but use the one-shot mode to open most urls.

Reply:
> True, a breaking change, but it breaks rather nicely in that users will get a clear error message.
I will accept your code, but I will modify it myself to make it work if given a single argument as before... as I think there's no reason for that to stop working..

> Here's a suggestion you can of course take or leave...

Yeah, that could be nice... I will try to get some time to make some improvementsl like this in the near future. Thanks!